src/ucs/sys/string.c

```diff
@@ -327,29 +327,45 @@ const char* ucs_flags_str(char *buf, size_t max,
 
 ssize_t ucs_path_calc_distance(const char *path1, const char *path2)
 {
-    unsigned distance = 0;
-    int same          = 1;
+    size_t i = 0;
+    unsigned distance;
     char resolved_path1[PATH_MAX], resolved_path2[PATH_MAX];
-    size_t comp_len, i;
+    char *larger_str;
+    size_t larger_len, smaller_len;
     size_t rp_len1, rp_len2;
 
     if ((NULL == realpath(path1, resolved_path1)) ||
         (NULL == realpath(path2, resolved_path2))) {
         return UCS_ERR_INVALID_PARAM;
     }
 
-    rp_len1  = strlen(resolved_path1);
-    rp_len2  = strlen(resolved_path2);
-    comp_len = ucs_min(rp_len1, rp_len2);
+    rp_len1 = strlen(resolved_path1);
+    rp_len2 = strlen(resolved_path2);
 
-    for (i = 0; i < comp_len; i++) {
-        if (resolved_path1[i] != resolved_path2[i]) {
-            same = 0;
-        }
+    if (rp_len1 > rp_len2) {
+        larger_len  = rp_len1;
+        smaller_len = rp_len2;
+        larger_str  = resolved_path1;
+    } else {
+        larger_len  = rp_len2;
+        smaller_len = rp_len1;
+        larger_str = resolved_path2;
+    }
+
+    /* find offset at which the paths diverge excluding the terminating
+       character */
+    while ((i < (smaller_len - 1)) &&
+           (resolved_path1[i] == resolved_path2[i])) {
+        i++;
+    }
```

Comment:
1. add helper function ucs_string_common_prefix_len(str1,str2) to return common length
2. add helper funciton ucs_string_count_char(str,char) to count how many times char appears in str
3. from this function, return max(count_char(str1+prefixlen,\), count_char(str2+prefixlen,\)

Reply:
@yosefe Thanks. I've addressed the above three now.